src/supertux/menu/keyboard_menu.cpp

```diff
@@ -109,37 +127,39 @@ KeyboardMenu::refresh()
   ItemControlField* micf;
 
   micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::UP)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::UP)));
+  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(m_player_id, Control::UP)));
   micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::DOWN)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::DOWN)));
+  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(m_player_id, Control::DOWN)));
   micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::LEFT)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::LEFT)));
+  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(m_player_id, Control::LEFT)));
   micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::RIGHT)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::RIGHT)));
+  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(m_player_id, Control::RIGHT)));
   micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::JUMP)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::JUMP)));
+  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(m_player_id, Control::JUMP)));
   micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::ACTION)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::ACTION)));
-  micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::PEEK_LEFT)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::PEEK_LEFT)));
-  micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::PEEK_RIGHT)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::PEEK_RIGHT)));
-  micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::PEEK_UP)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::PEEK_UP)));
-  micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::PEEK_DOWN)));
-  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::PEEK_DOWN)));
+  if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(m_player_id, Control::ACTION)));
 
-  if (g_config->developer_mode) {
-    micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::CHEAT_MENU)));
-    if (micf) micf->change_input(get_key_name(kbd_cfg.reversemap_key(Control::CHEAT_MENU)));
+  if (m_player_id == 0)
+  {
+    micf = dynamic_cast<ItemControlField*>(&get_item_by_id(static_cast<int>(Control::PEEK_LEFT)));
```

Comment:
Somehow, I'd love for this to be an array of the different controls and then iterating over that.

Reply:
Not all controls are rebindable; `ESCAPE`, `MENU_SELECT` and `MENU_SELECT_SPACE` aren't.